samples/authUserFlow.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+function main(projectId, clientId, clientSecret, refreshToken) {
+  // [START bigquery_auth_user_flow]
+  const {UserRefreshClient} = require('google-auth-library');
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const clientId = 'CLIENT ID'
+  // const clientSecret = 'CLIENT SECRET'
+  // const refreshToken = 'REFRESH TOKEN'
+  // const projectId = 'PROJECT ID'
+  const credentials = {
+    type: 'authorized_user',
+    client_id: clientId,
+    client_secret: clientSecret,
+    refresh_token: refreshToken,
+  };
+
+  const refreshClient = new UserRefreshClient();
+  refreshClient.fromJSON(credentials);
+  // [END bigquery_auth_user_flow]
+  // [START bigquery_auth_user_query]
+  async function query() {
+    const {BigQuery} = require('@google-cloud/bigquery');
+
+    const bigquery = new BigQuery({
+      projectId: projectId,
+    });
+
+    bigquery.authClient.cachedCredential = refreshClient;
```

Comment:
I believe we should support passing an auth client instance right to the BigQuery constructor:

```js
const bigquery = new BigQuery({
  authClient: refreshClient,
  projectId: projectId,
});
```

The common library is ready for this: https://github.com/googleapis/nodejs-common/blob/20f614a637a9573abe59b9fe6ec45e45cb59d03e/src/service.ts#L61, although it doesn't seem like the BigQuery constructor is allowing it to be done: https://github.com/googleapis/nodejs-bigquery/blob/6729fe8f0a13e660a289c641aae3b21411ed974e/src/bigquery.ts#L259 (it should be in `config`).

Reply:
I updated it to just passing the credentials to the constructor, per your later comment